src/main/java/hudson/plugins/openid/OpenIdSession.java

```diff
@@ -87,7 +89,7 @@ public HttpResponse doCommenceLogin() throws IOException, OpenIDException {
         String url = authReq.getDestinationUrl(true);
 
         // remember this in the session
-        Stapler.getCurrentRequest().getSession().setAttribute(SESSION_NAME,this);
+        Stapler.getCurrentRequest().getSession().setAttribute(SESSION_NAME,this.toString());
```

Comment:
Initial implementation causes "Store of non serializable object into HttpSession" and cannot be solved due to the nested classes from external libs. BTW, for me it smells like a typo in the code